test/sv/interfaces.mlir

```diff
@@ -0,0 +1,40 @@
+// RUN: circt-opt %s | FileCheck %s
+// RUN: circt-opt %s | circt-opt | FileCheck %s
+
+// CHECK-LABEL: func @test() {
+func @test() {
+  // Basic interface smoke test
+
+  sv.interface @myinterface {
```

Comment:
Where should these ops live?  I assume they would be top level, outside of modules/functions and other regions?

Reply:
I'm imagining them at the top level, alongside modules.

For example, we might want to declare an interface using this operation, and then somehow refer to the interface and one of its modports as the port list of a module. Mixing these ops and FIRRTL ops, for example, we could have something roughly like this:

```
firrtl.circuit "mycircuit" {
  sv.interface @myinterface {
    sv.interface.signal ...
    sv.interface.modport @mymodport ...
  }
  firrtl.module @mymodule(arg0: !sv.interface<@myinterface, @mymodport>) {
    ...
  }
}
```

I think there is a larger discussion to be had about exactly how that would work, but hopefully this illustrates how these new ops might fit in.

Reply:
This is the way it is done in SystemVerilog. Given that this dialect is to match SV as closely as possible I think this is the right way to model this.

Reply:
@lattner @teqdruid I've been working on a proof-of-concept based on this discussion and I've fleshed out some more ideas in here: https://llvm.discourse.group/t/firrtl-aggregate-types-discussion/1806/9.

Reply:
Cool, agreed, having them at the top level as peers to modules makes sense to me.